### ISSUE TITLE
Some calls to data endpoint return all zeros

### ISSUE THREAD

**Issue by basilveerman** [pacificclimate/climate-explorer-backend#15]
Some data api calls are still returning incorrect information.

Some of the requests that @jameshiebert and I noted as returning zeros now return [expected](http://docker1.pcic.uvic.ca:9000/api/data?model=CanESM2&variable=rx5dayETCCDI&emission=rcp85&area=&time=1) [data](http://docker1.pcic.uvic.ca:9000/api/data?model=CSIRO-Mk3-6-0&variable=rx1dayETCCDI&emission=rcp26&area=POLYGON+%28%28-128.90625+60.41015624999999%2C+-128.90625+63.73046874999999%2C+-121.875+63.73046874999999%2C+-121.875+60.41015624999999%2C+-128.90625+60.41015624999999%29%29&time=0), but other still don't.

Using a constant polygon:

``` bash
POLYGON="POLYGON+((-127.9296875+45.76171875%2C+-127.9296875+54.6484375%2C+-118.65234375000001+54.6484375%2C+-118.65234375000001+45.76171875%2C+-127.9296875+45.76171875))"
```

Works with January:

``` bash
curl -s 'http://docker1.pcic.uvic.ca:9000/api/data?model=CSIRO-Mk3-6-0&variable=rx5dayETCCDI&emission=rcp85&area='$POLYGON'&time=0' | python -m json.tool
```

``` json
{
    "r1i1p1": {
        "data": {
            "2025-01-16T00:00:00Z": 94.53663330078125,
            "2055-01-16T00:00:00Z": 101.20841064453126,
            "2085-01-16T00:00:00Z": 111.72054443359374
        },
        "units": "mm"
    },
    "r2i1p1": {
        "data": {
            "2025-01-16T00:00:00Z": 98.94828491210937,
            "2055-01-16T00:00:00Z": 102.28649291992187,
            "2085-01-16T00:00:00Z": 114.10343017578126
        },
        "units": "mm"
    },
    "r3i1p1": {
        "data": {
            "2025-01-16T00:00:00Z": 93.01469116210937,
            "2055-01-16T00:00:00Z": 103.7731689453125,
            "2085-01-16T00:00:00Z": 112.4370849609375
        },
        "units": "mm"
    },
    "r4i1p1": {
        "data": {
            "2025-01-16T00:00:00Z": 95.85071411132813,
            "2055-01-16T00:00:00Z": 102.0443115234375,
            "2085-01-16T00:00:00Z": 115.915771484375
        },
        "units": "mm"
    }
}
```

Not for any other timestep:

``` bash
curl -s 'http://docker1.pcic.uvic.ca:9000/api/data?model=CSIRO-Mk3-6-0&variable=rx5dayETCCDI&emission=rcp85&area='$POLYGON'&time=1' | python -m json.tool
```

``` json
{
    "r1i1p1": {
        "data": {
            "2025-02-15T00:00:00Z": 0.0,
            "2055-02-15T00:00:00Z": 0.0,
            "2085-02-15T00:00:00Z": 0.0
        },
        "units": "mm"
    },
    "r2i1p1": {
        "data": {
            "2025-02-15T00:00:00Z": 0.0,
            "2055-02-15T00:00:00Z": 0.0,
            "2085-02-15T00:00:00Z": 0.0
        },
        "units": "mm"
    },
    "r3i1p1": {
        "data": {
            "2025-02-15T00:00:00Z": 0.0,
            "2055-02-15T00:00:00Z": 0.0,
            "2085-02-15T00:00:00Z": 0.0
        },
        "units": "mm"
    },
    "r4i1p1": {
        "data": {
            "2025-02-15T00:00:00Z": 0.0,
            "2055-02-15T00:00:00Z": 0.0,
            "2085-02-15T00:00:00Z": 0.0
        },
        "units": "mm"
    }
}
```

Calls for the same model with rcp26 appear to work for alternate time of years:

``` bash
curl -s 'http://docker1.pcic.uvic.ca:9000/api/data?model=CSIRO-Mk3-6-0&variable=rx5dayETCCDI&emission=rcp26&area='$POLYGON'&time=1' | python -m json.tool
```

``` json
{
    "r1i1p1": {
        "data": {
            "2025-02-15T00:00:00Z": 55.08775024414062,
            "2055-02-15T00:00:00Z": 58.02188720703125,
            "2085-02-15T00:00:00Z": 54.56318969726563
        },
        "units": "mm"
    },
    "r2i1p1": {
        "data": {
            "2025-02-15T00:00:00Z": 59.523980712890626,
            "2055-02-15T00:00:00Z": 59.92850952148437,
            "2085-02-15T00:00:00Z": 58.3666015625
        },
        "units": "mm"
    },
    "r3i1p1": {
        "data": {
            "2025-02-15T00:00:00Z": 53.345208740234376,
            "2055-02-15T00:00:00Z": 57.764373779296875,
            "2085-02-15T00:00:00Z": 52.764739990234375
        },
        "units": "mm"
    },
    "r4i1p1": {
        "data": {
            "2025-02-15T00:00:00Z": 56.85017700195313,
            "2055-02-15T00:00:00Z": 59.44209594726563,
            "2085-02-15T00:00:00Z": 54.10572509765625
        },
        "units": "mm"
    },
    "r5i1p1": {
        "data": {
            "2025-02-15T00:00:00Z": 54.548590087890624,
            "2055-02-15T00:00:00Z": 0.0,
            "2085-02-15T00:00:00Z": 0.0
        },
        "units": "mm"
    }
}
```

But some returned results are 0 when they should not be.

The same issue persists for other variables:

``` bash
curl -s 'http://docker1.pcic.uvic.ca:9000/api/data?model=CSIRO-Mk3-6-0&variable=rx1dayETCCDI&emission=rcp26&area='$POLYGON'&time=1' | python -m json.tool
```

``` json
{
    "r1i1p1": {
        "data": {
            "2025-02-15T00:00:00Z": 19.7260986328125,
            "2055-02-15T00:00:00Z": 22.774565124511717,
            "2085-02-15T00:00:00Z": 21.199964904785155
        },
        "units": "mm"
    },
    "r2i1p1": {
        "data": {
            "2025-02-15T00:00:00Z": 20.879574584960938,
            "2055-02-15T00:00:00Z": 22.38013153076172,
            "2085-02-15T00:00:00Z": 21.877273559570312
        },
        "units": "mm"
    },
    "r3i1p1": {
        "data": {
            "2025-02-15T00:00:00Z": 21.111979675292968,
            "2055-02-15T00:00:00Z": 22.47863006591797,
            "2085-02-15T00:00:00Z": 19.134806823730468
        },
        "units": "mm"
    },
    "r4i1p1": {
        "data": {
            "2025-02-15T00:00:00Z": 0.0,
            "2055-02-15T00:00:00Z": 0.0,
            "2085-02-15T00:00:00Z": 0.0
        },
        "units": "mm"
    },
    "r5i1p1": {
        "data": {
            "2025-02-15T00:00:00Z": 0.0,
            "2055-02-15T00:00:00Z": 0.0,
            "2085-02-15T00:00:00Z": 0.0
        },
        "units": "mm"
    }
}
```

The back end does not show any errors for these requests.

**Comment by jameshiebert** [comment]
So it turns out that the problem is that our API is not specific enough to differentiate between climatological means of different time scales. You have noticed that this problem cropped up with rx5day and rx1day. That's because both of these are computed at monthly *and* annual scale. The former has timestep values greater than 0 (so you actually get data) while that latter does not (and comes back as zeros). Which you get, just depends on the ordering in the database.

This isn't just a problem for climdex, it will also be a problem when we start offering data at multiple timescales, so we should probably fix this in the API. We *do* have a [time resolution attribute](https://github.com/pacificclimate/modelmeta/blob/master/modelmeta/v2.py#L250) in the data model, so we should probably start actually using it.

**Comment by basilveerman** [comment]
I agree. We need to add a bit of functionality to the backend accounting for time resolution.

In this case however, the yearly timescale data is not present in the current `ce` ensemble and therefore should never be addressed by the backend. I removed them before the demo.

```sql
select ensemble_name, ensembles.ensemble_id, filename from ensembles natural join ensemble_data_file_variables natural join data_file_variables natural join data_files where ensemble_name = 'ce' and (filename ilike '%ayr%' or filename ilike '%yrclim%');
```

```
 ensemble_name | ensemble_id | filename 
---------------+-------------+----------
(0 rows)
```

You motivated me to look into it a bit more from that perspective however and I found that we don't filter on ensemble_id in the `data` [api endpoint](https://github.com/pacificclimate/climate-explorer-backend/blob/55315b8e1a7b18d44f4142ac23c5120cd7544864/ce/api/data.py#L75).

Adding the yrClim datasets back in will require we also account for multiple timescales. They are currently in the ensemble named `ce_testing`.
